components/scream/src/physics/p3/tests/p3_run_and_cmp.cpp

```diff
@@ -93,13 +93,16 @@ static Int compare (const std::string& label, const Scalar* a,
 }
 
 struct Baseline {
-  Baseline (const Int nsteps=6, const Int ncol=3, const Int repeat=0, const std::string predict_nc="both") :
+  Baseline (const Int nsteps=6, const Int ncol=3, const Int repeat=0, const std::string predict_nc="both", const std::string prescribed_ccn="both") :
     m_ic_ncol(ncol), m_repeat(repeat)
   {
-    for (const bool do_predict_nc : {true, false}) {
-      if (predict_nc == "both" || ( (do_predict_nc and predict_nc == "yes") || (!do_predict_nc and predict_nc == "no") )) {
-        //                 initial condit,     dt,  nsteps, prescribe or predict nc
-        params_.push_back({ic::Factory::mixed, 300, nsteps, do_predict_nc});
+    for (const bool do_prescribed_CCN : {true, false}) {
+      for (const bool do_predict_nc : {true, false}) {
+        if ((predict_nc == "both"    || ( (do_predict_nc and predict_nc == "yes")         || (!do_predict_nc and predict_nc == "no") )) || 
+           (prescribed_ccn == "both" || ( (do_prescribed_CCN and prescribed_ccn == "yes") || (!do_prescribed_CCN and prescribed_ccn == "no") ))) {
+          //                 initial condit,     dt,  nsteps, prescribe or predict nc, prescribe CCN or not
+          params_.push_back({ic::Factory::mixed, 300, nsteps, do_predict_nc, do_prescribed_CCN});
+        }
```

Comment:
what happens when none of these options are true.  Does params_ just not add a new case?  Note, I think this question would go back to the original code before this PR as well.

Reply:
I think you're right to question that. I rewrote the logic to solve the problem you're mentioning...